bindings/python/tests/test_jstore.py

```diff
@@ -0,0 +1,206 @@
+"""Test JStore."""
+from pathlib import Path
+
+import dlite
+from dlite.testutils import raises
+
+
+thisdir = Path(__file__).resolve().parent
+outdir = thisdir / "output"
+indir = thisdir / "input"
+entitydir = thisdir / "entities"
+
+dlite.storage_path.append(entitydir)
+
+
+# Test format_dict(), arg: soft7
+D1 = {
+    "uri": "http://onto-ns.com/meta/ex/0.2/Test",
+    "dimensions": {"n": "number of something"},
+    "properties": {
+        "a": {"type": "string"},
+        "b": {"type": "float64", "shape": ["n"]},
+    },
+}
+D2 = {
+    "uri": "http://onto-ns.com/meta/ex/0.2/Test",
+    "dimensions": [{"name": "n", "description": "number of something"}],
+    "properties": [
+        {"name": "a", "type": "string"},
+        {"name": "b", "type": "float64", "shape": ["n"]},
+    ],
+}
+assert dlite.format_dict(D1, soft7=True) == D1
+assert dlite.format_dict(D1, soft7=False) == D2
+assert dlite.format_dict(D2, soft7=True) == D1
+assert dlite.format_dict(D2, soft7=False) == D2
+
+d1 = {
+    "uuid": "d6a1c1db-44b6-5b87-b815-83f1127395b6",
+    "meta": "http://onto-ns.com/meta/ex/0.2/Test",
+    "dimensions": {"n": 3},
+    "properties": {
+        "a": "hello",
+        "b": [1.1, 2.2, 3.3],
+    },
+}
+d2 = {
+    "uuid": "d6a1c1db-44b6-5b87-b815-83f1127395b6",
+    "meta": "http://onto-ns.com/meta/ex/0.2/Test",
+    "dimensions": {"n": 3},
+    "properties": {
+        "a": "hello",
+        "b": [1.1, 2.2, 3.3],
+    },
+}
```

Comment:
What is the difference between these two?

Reply:
Good observation. You are right, for data instances there are no difference between the soft5 and soft7 representation. I added a comment about this to the test